cmd/root.go

```diff
@@ -314,6 +312,38 @@ func parseConfig(cmd *cobra.Command, conf *proxy.Config, args []string) error {
 	return nil
 }
 
+// parseBoolOpt parses a boolean option from the query string
+//  name the name of the option in the query string
+//  q the map of the parsed query string parameters from url.ParseQuery()
+// returns
+//   nil if no value was found
+//   true if the value is 't' or 'true' case insensitive
+//   false if the value is 'f' or 'false' case insensitive
+func parseBoolOpt(name string, q map[string][]string) (*bool, error) {
+	if iam, ok := q[name]; ok {
+		if len(iam) != 1 {
+			return nil, newBadCommandError(fmt.Sprintf("auto iam authn param should be only one value: %q", iam))
```

Comment:
need to generalize this beyond "auto-iam-authn"

Reply:
done